__tests__/integration/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,257 @@
+        const redeemScript = walletUtils.createP2SHRedeemScript(
+          pubkeys,
+          params.multisig.minSignatures,
+          i
+        );
+        const address = Address.payingTo(Script.fromBuffer(redeemScript), NETWORK_NAME);
+        addressesArray.push(address.toString());
+
+        // Informing debug data
+        multisigDebugData = {
+          total: pubkeys.length,
+          minSignatures: params.multisig.minSignatures,
+          pubkeys,
+        };
+      }
+    } else {
+      // Generating 24-word seed if none was informed
+      if (!wordsInput) {
+        wordsInput = wallet.generateWalletWords();
+      }
+
+      /*
+       * Since the objective of this script is to mimic the addresses generated on a simple `/start`
+       * request for the Wallet Headless, we use this standard derivation index.
+       * Future changes on this walletUtils method may cause this step to produce different results.
+       */
+      const accountDerivationIndex = '0\'/0';
+
+      // Common address calculation
+      const xpubkey = walletUtils.getXPubKeyFromSeed(wordsInput, {
+        networkName: NETWORK_NAME,
+        accountDerivationIndex,
+      });
+      const addresses = walletUtils.getAddresses(
+        xpubkey,
+        addressIntervalStart,
+        addressIntervalEnd,
+        NETWORK_NAME
+      );
+
+      // Formatting addresses to a simple array format
+      for (const hash in addresses) {
+        addressesArray[addresses[hash]] = hash;
+      }
+    }
+
+    // Finishing benchmark and returning results
+    const timeEnd = Date.now().valueOf();
+    const timeDiff = timeEnd - timeStart;
+
+    const returnObject = {
+      duration: timeDiff,
+      isUsed: false,
+      words: wordsInput,
+      addresses: addressesArray
+    };
+    if (params.multisig) {
+      returnObject.multisigDebugData = multisigDebugData;
+    }
+    return returnObject;
+  }
+
+  /**
+   * Reads a JSON file containing wallets and parses it
+   * @returns {Promise<any>}
+   * @throws SyntaxError
+   * @private
+   */
+  async _deserializeWalletsFile() {
+    const dataBuffer = await fs.readFile(this.WALLETS_FILENAME);
+    const strData = dataBuffer.toString();
+
+    try {
+      const jsonData = JSON.parse(strData);
+      return jsonData;
+    } catch (err) {
+      console.error('Corrupt wallets file');
+      throw err;
+    }
+  }
+
+  /**
+   * Writes the contents of a wallet array in a human-readable way, but with one wallet per line
+   * @param {unknown[]} wallets
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _serializeWalletsFile(wallets) {
+    /*
+     * The main aim of this file structure is human readability for debugging.
+     * The result must be a valid JSON, but with only one line per wallet.
+     */
+    let strWalletsData = wallets
+      .map(w => `${JSON.stringify(w)}`)
+      .join(',\n');
+    strWalletsData = `[\n${strWalletsData}\n]`;
```

Comment:
The original assumption for this helper, on HathorNetwork/hathor-wallet-headless#189 , was to have one wallet per line on the JSON. Hence this manual serialization.

The use of `JSON.stringify(wallets,null,1)` would make each wallet have around 30 lines. It is a perfectly valid change, but I feel this would be better done on a separate issue since it would have little impact on this PR's objective.

Do you agree?